apollo-router/src/plugins/telemetry/tracing/apollo_telemetry.rs

```diff
@@ -128,12 +128,19 @@ struct LightSpanData {
 
 impl From<SpanData> for LightSpanData {
     fn from(value: SpanData) -> Self {
+        let attr_len = value.attributes.len();
         Self {
             span_id: value.span_context.span_id(),
             name: value.name,
             start_time: value.start_time,
             end_time: value.end_time,
-            attributes: value.attributes,
+            attributes: value.attributes.into_iter().fold(
```

Comment:
This looks like an optimization to save us just copying across all the attributes. Is that correct?

Reply:
`value.attributes` is now a `Vec`, so all the `get` calls we do would be much less effecient as `.iter().find(...)`. Could consider a different solution but this was the smallest change.